tabular/src/autogluon/tabular/models/tabular_nn/torch/tabular_nn_torch.py

```diff
@@ -562,9 +562,34 @@ def _get_default_resources(self):
         num_gpus = 0
         return num_cpus, num_gpus
 
+    def save(self, path: str = None, verbose=True) -> str:
+        import torch
+        # Save on CPU to ensure the model can be loaded on a box without GPU
+        self.model = self.model.to(torch.device("cpu"))
+        path = super().save(path, verbose)
+        # Put the model back to the device after the save
+        self.model.to(self.device)
+        return path
+
     @classmethod
     def load(cls, path: str, reset_paths=True, verbose=True):
+        import torch
```

Comment:
Did you check other models to see if they have this problem? (ex: FastAI)

Reply:
Yes, I did also trained the full stack (`'RandomForestEntr', 'LightGBM', 'WeightedEnsemble_L2', 'NeuralNetFastAI', 'KNeighborsUnif', 'CatBoost', 'RandomForestGini', 'KNeighborsDist', 'XGBoost', 'LightGBMLarge', 'ExtraTreesEntr', 'NeuralNetTorch', 'LightGBMXT', 'ExtraTreesGini'`) with:

```python
predictor = TabularPredictor(label=label).fit(train_data[100:], num_gpus=1)
```

No other models had issues with this setup.